ui/src/app/components/genomics-extraction-table.tsx

```diff
@@ -218,114 +216,127 @@ const FailedRequestMessage = () => (
-              >
-                <Column
-                  header='Dataset Name'
-                  field='datasetNameDisplay'
-                  sortable
-                  sortField='datasetName'
-                  style={{
-                    maxWidth: '8rem',
-                    textOverflow: 'ellipsis',
-                    overflow: 'hidden',
-                    whiteSpace: 'nowrap',
-                  }}
-                />
-                <Column
-                  header='Status'
-                  field='statusDisplay'
-                  sortable
-                  sortField='statusOrdinal'
-                />
-                <Column
-                  header='Date Started'
-                  field='dateStartedDisplay'
-                  sortable
-                  sortField='dateStarted'
-                />
-                <Column
-                  header='Cost'
-                  field='costDisplay'
-                  sortable
-                  sortField='cost'
-                />
-                <Column
-                  header='Size'
-                  field='sizeDisplay'
-                  sortable
-                  sortField='size'
-                />
-                <Column
-                  header='Duration'
-                  field='durationDisplay'
-                  sortable
-                  sortField='duration'
-                />
-                <Column header='' field='menuJsx' />
-              </DataTable>
-            )}
-          </CSSTransition>
-        </SwitchTransition>
+    return (
+      <div
+        id='extraction-data-table-container'
+        className='extraction-data-table-container'
+      >
+        <div className='slim-scroll-bar'>
+          <SwitchTransition>
+            <CSSTransition
+              key={!jobs}
+              classNames='switch-transition'
+              addEndListener={(node, done) => {
+                node.addEventListener(
+                  'transitionend',
+                  (e) => {
+                    if (node.isEqualNode(e.target)) {
+                      done(e);
+                    }
+                  },
+                  false
+                );
+              }}
+            >
+              {!jobs ? (
+                <Spinner style={{ display: 'block', margin: '3rem auto' }} />
+              ) : error ? (
+                <FailedRequestMessage />
+              ) : (
+                <DataTable
+                  autoLayout
+                  emptyMessage={<EmptyTableMessage />}
+                  sortField={!jobs || jobs.length > 0 ? 'dateStarted' : ''}
+                  sortOrder={-1}
+                  value={jobs.map((job) =>
+                    mapJobToTableRow(
+                      job,
+                      workspace,
+                      (updateFn, optimisticValue) => {
+                        // Update the extraction job list optimistically.
+                        const optimisticData = jobs.map((j) => {
```

Comment:
Optimistic data update is here. It works, but unfortunately it's not well suited to this use case (see PR description).